api/incubator/src/main/java/io/opentelemetry/api/incubator/config/ConfigProvider.java

```diff
@@ -30,6 +32,33 @@ public interface ConfigProvider {
   @Nullable
   DeclarativeConfigProperties getInstrumentationConfig();
 
+  /**
+   * Returns the {@link DeclarativeConfigProperties} for a specific instrumentation by name. If no
+   * configuration is available for the given name, an empty {@link DeclarativeConfigProperties} is
+   * returned.
+   *
+   * @param name the name of the instrumentation
+   * @return the {@link DeclarativeConfigProperties} for the given instrumentation name
+   */
+  default DeclarativeConfigProperties getJavaInstrumentationConfig(String name) {
+    DeclarativeConfigProperties config = getInstrumentationConfig();
+    return config == null ? empty() : config.get("java").get(name);
+  }
+
+  /**
+   * Returns the {@link DeclarativeConfigProperties} for general instrumentation config by name. If
+   * no configuration is available for the given name, an empty {@link DeclarativeConfigProperties}
+   * is returned.
+   *
+   * @param name the name of the general instrumentation config
+   * @return the {@link DeclarativeConfigProperties} for the given general instrumentation config
+   *     name
+   */
+  default DeclarativeConfigProperties getGeneralInstrumentationConfig(String name) {
```

Comment:
This works because for now there are no top level properties under `.instrumentation/development.general`. But if there were, callers would be unable to access them. I think we should omit the `String name` parameter for now and just return the `.instrumentation/development.general` node.

Reply:
This is only a shortcut method - so I think we're allowed to nudge towards what we think is best practice. 
I think that is the currently implemented solution, but I can live with either.

Reply:
> oh good point, can .instrumentation/development.java. have top-level properties?

No - its properties are limited to objects: https://github.com/open-telemetry/opentelemetry-configuration/blob/main/schema-docs.md#experimentallanguagespecificinstrumentation-

In contrast, `.general` has explicit properties of `http` and `peer` which are objects, but nothing stating all properties are objects: https://github.com/open-telemetry/opentelemetry-configuration/blob/main/schema-docs.md#experimentalgeneralinstrumentation-

Reply:
ok, I'll adapt "general" then

Reply:
done